src/coreclr/jit/emit.cpp

```diff
@@ -1006,14 +1006,19 @@ insGroup* emitter::emitSavIG(bool emitAdd)
         assert(emitLastIns != nullptr);
         assert(emitCurIGfreeBase <= (BYTE*)emitLastIns);
         assert((BYTE*)emitLastIns < emitCurIGfreeBase + sz);
-
-#if defined(TARGET_XARCH)
         assert(emitLastIns != nullptr);
-        if (emitLastIns->idIns() == INS_jmp)
+
+#ifdef TARGET_XARCH
+        instruction jmpInstr = INS_jmp;
+#else // TARGET_ARM64
+        instruction jmpInstr = INS_b;
+#endif
```

Comment:
Won't the assumption that it's either ARM or XARCH be invalid on LoongArch64 builds?

Reply:
they will be and I need to do fix/validate it. This is draft and I would like to get a sense of its state before I spend time on other archs.